src/pages/register-page/register-page.tsx

```diff
@@ -23,16 +44,24 @@ export const RegisterPage = () => {
         {messages.alreadyHaveAnAccountText}
         <Link to={PagePath.loginPage}>{messages.buttons.signIn}</Link>
       </p>
-      <form action="" className={styles.formContainer}>
+      <form
+        action=""
+        className={styles.formContainer}
+        onSubmit={handleSubmit(onSubmit)}
+      >
         {FIELDS.map(field => {
+          const error = errors[field.name]?.message;
+          const rules = validationRules[field.name];
+
           if (field.type === 'country-select') {
             return (
               <CountrySelect
                 key={field.name}
-                name={field.name}
                 label={field.label}
                 options={countryOptions}
                 className={styles.formInput}
+                {...register(field.name, rules)}
+                error={typeof error === 'string' ? error : undefined}
```

Comment:
```suggestion
                error={error}
```
потому что судя по твоему коду у тебя error и так либо строка либо undefined